### DISCUSSION TITLE
DAC on the crazyflie

### DISCUSSION THREAD

**Discussion by SriramKodey** [bitcraze/discussions#1451] [Q&A] [answered]
Hi, I am trying to use the DAC peripheral on the crazyflie. The default Makefile includes the stm32 standard peripheral library, however the build process gives the following errors:

/usr/lib/gcc/arm-none-eabi/10.3.1/../../../arm-none-eabi/bin/ld: src/built-in.o: in function `dacInit':
/home/r/crazyflie-firmware/build/../src/deck/api/deck_analog.c:41: undefined reference to `DAC_DeInit'
collect2: error: ld returned 1 exit status
make[1]: *** [tools/kbuild/Makefile.kbuild:570: firmware] Error 1
make[1]: Leaving directory '/home/r/crazyflie-firmware/build'
make: *** [tools/kbuild/Makefile.kbuild:150: sub-make] Error 2

Any insight is appreciated. Thanks!

**Comment by ataffanel** [comment, marked as answer]
The DAC driver does not seems to be in the build list. Can you try to add it and see if it solves the problem:

```diff
diff --git a/src/lib/Kbuild b/src/lib/Kbuild
index b2698d35..0ada3835 100644
--- a/src/lib/Kbuild
+++ b/src/lib/Kbuild
@@ -7,6 +7,7 @@ obj-y += FatFS/ffunicode.o
 
 # STM32F4xx_StdPeriph_Driver
 obj-y += STM32F4xx_StdPeriph_Driver/src/stm32f4xx_adc.o
+obj-y += STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dac.o
 obj-y += STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dbgmcu.o
 obj-y += STM32F4xx_StdPeriph_Driver/src/stm32f4xx_dma.o
 obj-y += STM32F4xx_StdPeriph_Driver/src/stm32f4xx_exti.o
```

If it does, this is something we will need to add in the repos. Please tell us if it work or, if you want, propose a PR with the change :-).
